src/08-recommendations/03-pipeline-recommend.ts

```diff
@@ -0,0 +1,325 @@
+/*
+ * This variation tries the same technique of feeding candidates to the LLM to have it
+ * choose and justify recommendations, but it differs from the previous attempts in two ways:
+ *
+ * - it sources the artwork candidates from our actual recommendations pipeline via a tool call
+ * - it provides fewer candidates
+ *
+ * Because this makes a real query to Metaphysics for artwork recommendations, it requires
+ * a valid user ID and access token to be provided as env vars / in the .env file.
+ */
+
+import OpenAI from "openai"
+import dotenv from "dotenv"
+import chalk from "chalk"
+
+dotenv.config()
+const openai = new OpenAI()
+
+const userID = process.env.RECOMMENDATIONS_USER_ID
+const accessToken = process.env.RECOMMENDATIONS_USER_ACCESS_TOKEN
```

Comment:
While trying this out I tried to pre-generate tokens for a few users (this team basically) so that I could try this out with a few different sets of candidate recs — but the tokens I generated in the console seemed to be considered invalid for API requests. I might need a refresher on how to do that.